.github/scripts/check-version.sh

```diff
@@ -0,0 +1,83 @@
+#!/bin/bash
+
+# Version checking script for Homebrew tap updates
+# Usage: check-version.sh <product_name> <incoming_version> <is_cask>
+# Outputs: should_update (true/false), current_version, incoming_version
+
+set -euo pipefail
+
+# Check arguments
+if [[ $# -ne 3 ]]; then
+    echo "Usage: $0 <product_name> <incoming_version> <is_cask>" >&2
+    exit 1
+fi
+
+PRODUCT_NAME="$1"
+INCOMING_VERSION="$2"
+IS_CASK="$3"
+
+# Determine file path based on whether it's a cask or formula
+if [[ "$IS_CASK" == "true" ]]; then
+    FILE_PATH="./Casks/hashi-${PRODUCT_NAME}.rb"
+else
+    FILE_PATH="./Formula/${PRODUCT_NAME}.rb"
+fi
+
+# Check if the file exists
+if [[ ! -f "$FILE_PATH" ]]; then
+    echo "File $FILE_PATH does not exist. This is a new product, proceeding with update." >&2
```

Comment:
🎨 optionally, you might want to use a function for reporting errors or log messages.  This avoids forgetting to redirect the output when adding a new message.  For example, you could copy any or all of [this blob](https://github.com/hashicorp-forge/actions-pao-tool/blob/main/scripts/utils.bash#L3-L8) into the script.  The error count stuff is probably not helpful, but all your "echo ... ; exit 1" calls could become "die" calls, and all the other "echo >&2" calls could call "log".  This also highlights for anyone reading the code that the output is on a separate stream from the key=value outputs emitted for caller use.